File: kotlin/PalindromeChecker.kt
fun isPalindrome(text: String): Boolean {
    var left = 0
    var right = text.length - 1

    while (left < right) {
        if (text[left] != text[right]) {
            return false
        }
        left++
        right--
    }

    return true
}

fun main() {
    print("Enter a string to check if it's a palindrome: ")
    val input = readLine().orEmpty()
    val processed = input.replace("\\s+".toRegex(), "").lowercase()

    if (isPalindrome(processed)) {
        println("\"$input\" is a palindrome.")
    } else {
        println("\"$input\" is not a palindrome.")
    }
}

File: swift/PalindromeChecker.swift
import Foundation

func isPalindrome(_ text: String) -> Bool {
    let characters = Array(text)
    var left = 0
    var right = characters.count - 1

    while left < right {
        if characters[left] != characters[right] {
            return false
        }
        left += 1
        right -= 1
    }

    return true
}

print("Enter a string to check if it's a palindrome: ", terminator: "")
let input = readLine() ?? ""
let processed = input.replacingOccurrences(of: "\\s+", with: "", options: .regularExpression).lowercased()

if isPalindrome(processed) {
    print("\"\(input)\" is a palindrome.")
} else {
    print("\"\(input)\" is not a palindrome.")
}
